### PR TITLE
Minor changes for py3.8 to comply with issue34850

### PR DESCRIPTION
**Pull request by andre-geldenhuis**
There are some warnings generated when using python 3.8 due to [issue34850](https://bugs.python.org/issue34850).

I made some very minor syntax changes to address these warnings.

### CODE REVIEW

**Comment by andre-geldenhuis** [comment]
The variables outOfRailState, apogeeState, impactState are sometimes boolean and sometimes an array.  With python 3.8 this causes an error as the truth value of an array with more than one element is ambiguous.

I initialised the variables as single element arrays containing 0 and testing if all the values inside are 0 as this keeps the logic similar to before.  However, I'm not sure that's the clearest approach.  It might be clearer to test the array length?

**Comment by giovaniceotto** [comment]
Thanks for bringing up this issue!

I agree with you, perhaps checking the length of the array would be the best option to solve the issue. I ran some speed tests to verify if there is any significant difference between them in terms of execution time and this is the result:

```[python]
>>> timeit.timeit(stmt='a==0', setup='a=0') 
0.047009199999997975

>>> timeit.timeit(stmt='a is 0', setup='a=0') 
<timeit-src>:2: SyntaxWarning: "is" with a literal. Did you mean "=="?
<timeit-src>:6: SyntaxWarning: "is" with a literal. Did you mean "=="?
0.04326839999999521

>>> timeit.timeit(stmt='len(a) == 0', setup='import numpy as np; a = np.array([])')
0.08204170000000488

>>> timeit.timeit(stmt='(a==0).all()', setup='import numpy as np; a = np.array([0])') 
3.8522727000000145
```

As we can see, it appears that using `(a==0).all()` is significantly more time consuming. Doing some quick calculations, estimating that each flight simulation can reach up to 10k iterations, checking two if conditions like this, using `(a==0).all()` instead of `len(a) == 0` could increase simulation time by 100 ms.

What do you think @andre-geldenhuis?

File: rocketpy/Flight.py [hunk 1]
```diff
@@ -787,7 +787,7 @@ def __init__(
                     # print('\t\t\tAltitude: ', phase.solver.y[2])
                     # print('\t\t\tEvals: ', self.functionEvaluationsPerTimeStep[-1])
                     # Check for out of rail event
-                    if self.outOfRailState is 0 and (
+                    if (self.outOfRailState == 0).all() and (
```
Comment (giovaniceotto) [inline rocketpy/Flight.py:822]:
```suggestion
                    if len(self.outOfRailState) == 1 and (
```

File: rocketpy/Flight.py [hunk 2]
```diff
@@ -869,7 +869,7 @@ def __init__(
                         phase.timeNodes.addNode(self.t, [], [])
                         phase.solver.status = "finished"
                     # Check for apogee event
-                    if self.apogeeState is 0 and self.y[5] < 0:
+                    if (self.apogeeState == 0).all() and self.y[5] < 0:
```
Comment (giovaniceotto) [inline rocketpy/Flight.py:905]:
```suggestion
                    if len(self.apogeeState) == 1 and self.y[5] < 0:
```

File: rocketpy/Flight.py [hunk 3]
```diff
@@ -2016,7 +2016,7 @@ def info(self):
             print(name + " Parachute Inflated at Height of: {:.3f} m".format(altitude))
 
         # Print impact conditions
-        if not (self.impactState is 0):
+        if (self.impactState != 0).all():
```
Comment (giovaniceotto) [inline rocketpy/Flight.py:2113]:
```suggestion
        if len(self.impactState) != 0:
```

**Pull request merged**